lib/cache.js

```diff
@@ -79,5 +78,17 @@ var save = cache.save = function (cb) {
  * @return {Promise}
  */
 cache.clear = function (cb) {
-  return unlinkAsync(cache.source).nodeify(cb);
+  fs.unlink(cache.source, function (err) {
+    // if we weren't given a callback, it doesn't matter
+    if (!cb) {
+      return;
+    }
+
+    // swalling the error if it's a file does not exist error
+    if (err && err.code === 'ENOENT') {
+      return cb();
```

Comment:
This seems really brittle.  The only time `cb` is invoked explicitly without arguments is when there is an `ENOENT` error.  The "success" path on this function relies on line `err` being undefined on line 92, which really seems like this would be one future change away from introducing a bug where a success callback is never invoked.

Following the "fail early" philosophy would help here.  In that case, you'd examine `err` for conditions of failure and then go ahead and fail in that event, and then line 92 would be the happy success path.